src/main/java/kaiyap/KaiYap.java

```diff
@@ -0,0 +1,214 @@
+package kaiyap;
+
+import exceptions.AlreadyExistsException;
+import exceptions.InvalidInputException;
+import exceptions.KaiYapException;
+import exceptions.MissingInputException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class KaiYap {
+
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Parser parser;
+
+    public enum TaskType {
+        TODO, DEADLINE, EVENT
+    }
+
+    public KaiYap() {
+        this.ui = new Ui();
+        this.taskList = new TaskList(ui);
+        this.ui.setTaskList(this.taskList);
+        this.storage = new Storage(ui, taskList, System.getProperty("user.home") + "/data/", "kaiyap.txt");
+        this.parser = new Parser();
+    }
+
+    /**
+     * Starts the chat interface for the KaiYap application.
+     * This method initiates a loop that continuously reads user input from the console,
+     * processes it, and performs actions such as listing tasks, marking tasks as done or undone,
+     * deleting tasks, or adding new tasks. The loop terminates when the user inputs "bye".
+     */
+    public void startChat() {
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine().strip();
+        while (!input.equals("bye")) {
+            String action = parser.decideAction(input);
+            switch (action) {
+            case "listInputs":
+                ui.listInputs();
+                break;
+            case "mark":
+                try {
+                    this.markAsDone(input);
+                } catch (KaiYapException e) {
+                    ui.printError(e.getMessage());
+                }
+                break;
+            case "unmark":
+                try {
+                    this.unmarkDone(input);
+                } catch (KaiYapException e) {
+                    ui.printError(e.getMessage());
+                }
+                break;
+            case "find":
+                try {
```

Comment:
Instead of try catching every switch case, perhaps can wrap the entire switch statement in a single try catch? (still within the while loop)